projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js

```diff
@@ -2,32 +2,77 @@ import { getRedirectUrl } from '@automattic/jetpack-components';
 import { getSiteFragment, useAnalytics } from '@automattic/jetpack-shared-extension-utils';
 import { Modal, Button, CheckboxControl } from '@wordpress/components';
 import { usePrevious } from '@wordpress/compose';
-import { useSelect } from '@wordpress/data';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { store as editorStore } from '@wordpress/editor';
-import { useEffect, useState } from '@wordpress/element';
+import { useEffect, useRef, useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import './editor.scss';
 
 export const name = 'launchpad-save-modal';
 
 export const settings = {
+	// TODO: Rename LaunchpadSaveModal to something more generic
 	render: function LaunchpadSaveModal() {
-		const isSavingSite = useSelect(
-			select => select( editorStore ).isSavingNonPostEntityChanges(),
-			[]
-		);
-		const isSavingPost = useSelect( select => select( editorStore ).isSavingPost(), [] );
+		// First Post Published Modal
+		const postType = useSelect( select => select( editorStore ).getCurrentPostType(), [] );
+		const { link } = useSelect( select => select( 'core/editor' ).getCurrentPost() );
 		const isCurrentPostPublished = useSelect(
 			select => select( editorStore ).isCurrentPostPublished(),
 			[]
 		);
+		const shouldShowFirstPostPublishedModal = useSelect( select =>
```

Comment:
**NOTE:** 

L16-61 All of this is duplicated from the first post published modal in Calypso